controllers/temporalworkerprocess_controller.go

```diff
@@ -77,6 +87,73 @@ func (r *TemporalWorkerProcessReconciler) Reconcile(ctx context.Context, req ctr
 		return reconcile.Result{}, nil
 	}
 
+	if worker.Spec.Builder.BuilderEnabled() {
```

Comment:
Here it's a complete copy/past from the persistence reconciling. 
We could create a package which contains this code.
It will remove this code duplication and will allow us to simply unit testing :)

Reply:
I moved the job creation to a function and created job builder under resources.

How about you refactor the persistenct reconciler, building on what I started? I will then adapt the job in the temporalworkerprocess to use whatever you do to refactor that logic. Since there are far more jobs/complexity in persistence reconcile, it makes sense to refactor that first and do separate of this PR since that will force this PR to expand in other areas not originally intended to cover.

Reply:
Oh! I saw what you did. I was not thinking about the jobs themselves but in the whole job creation/check logic. And then both of the persistence jobs and worker jobs reconciliers would us this logic.